src/clj_kondo/impl/analyzer/spec.clj

```diff
@@ -31,6 +31,11 @@
                                                      (str "Unresolved symbol: " sym)))))))
     (analyze-children ctx body)))
 
+(defn analyze-def [{:keys [analyze-children resolved] :as ctx} expr]
+  (let [[kw-expr & body] (next (:children expr))]
+    (let [fq-def (symbol (-> resolved first name) (-> resolved second name))]
+      (analyze-children ctx (cons (assoc kw-expr :def fq-def) body)))))
+
```

Comment:
@borkdude I'm not sure if there's a better way to get the fully qualified name of the call?

Reply:
If you are using clj-kondo you will notice you have a redundant let here ;) 

I'll check out the branch and will make some changes locally, to play with it. Would you mind giving me push rights to this PR branch, perhaps?

Reply:
@snoe Instead of passing `analyze-children` to this function, you can now use `clj-kondo.impl.analyzer.common/analyze-children`, this is set up in a way to avoid circular dependencies.

Reply:
it's hooked up for both `clojure.spec.alpha/def` and `cljs.spec.alpha/def`, I suppose the clojure version might be fine?

I didn't realize it could take a symbol, I'll update, thanks.

Reply:
I split the `case` to just pass in the literals, I believe this should be good now.